### PR TITLE
Chef-16 fixes

### PR DESCRIPTION
**Pull request by lamont-granquist**
Fixes knife-ec2 for the knife config changes in Chef-16 + Chef >= 15.11.x

### CODE REVIEW

File: knife-ec2.gemspec [hunk 1]
```diff
@@ -13,9 +13,9 @@ Gem::Specification.new do |s|
   s.license      = "Apache-2.0"
 
   s.files        = %w{LICENSE} + Dir.glob("lib/**/*")
-  s.required_ruby_version = ">= 2.5"
+  s.required_ruby_version = ">= 2.6"
 
-  s.add_dependency "chef", ">= 15.1"
+  s.add_dependency "chef", ">= 15.10.21" # needs this version for Chef 16 backports
```
Comment (tas50) [inline knife-ec2.gemspec:18]: This should just be 15.11 since 15.10.21 never shipped

File: lib/chef/knife/ec2_server_create.rb [hunk 2]
```diff
@@ -62,14 +60,14 @@ class Ec2ServerCreate < Chef::Knife::Bootstrap
         description: "The security groups for this server; not allowed when using VPC",
         proc: Proc.new { |groups| groups.split(",") }
 
-      option :security_group_id,
+      option :security_group_ids,
         short: "-g SECURITY_GROUP_ID",
         long: "--security-group-id ID",
```
Comment (tas50) [inline lib/chef/knife/ec2_server_create.rb:65]: The long needs to be updated here as well if we're going to change the name

File: lib/chef/knife/ec2_server_create.rb [hunk 3]
```diff
@@ -62,14 +60,14 @@ class Ec2ServerCreate < Chef::Knife::Bootstrap
         description: "The security groups for this server; not allowed when using VPC",
         proc: Proc.new { |groups| groups.split(",") }
 
-      option :security_group_id,
+      option :security_group_ids,
```
Comment (tas50) [inline lib/chef/knife/ec2_server_create.rb:63]: Do we even need to change this if it just accumulates it up?

File: lib/chef/knife/ec2_server_create.rb [hunk 4]
```diff
@@ -524,42 +509,42 @@ def validation_key_tmpfile
...
         end
       end
 
       def bootstrap_common_params
-        config[:encrypted_data_bag_secret] = s3_secret || config_value(:secret)
-        config[:encrypted_data_bag_secret_file] = config_value(:secret_file)
+        config[:encrypted_data_bag_secret] = s3_secret || config[:secret]
+        config[:encrypted_data_bag_secret_file] = config[:secret_file]
         # retrieving the secret from S3 is unique to knife-ec2, so we need to set "command line secret" to the value fetched from S3
         # When linux vm is spawned, the chef's secret option proc function sets the value "command line secret" and this value is used by
```
Comment (tas50) [inline lib/chef/knife/ec2_server_create.rb:539]: Is this comment still valid?

File: lib/chef/knife/ec2_server_create.rb [hunk 5]
```diff
@@ -666,34 +651,34 @@ def plugin_validate_options!
...
+        if config[:ebs_encrypted]
           error_message = ""
           errors = []
           # validation for flavor and ebs_encrypted
-          if !config_value(:flavor)
+          if !config[:flavor]
             ui.error("--ebs-encrypted option requires valid flavor to be specified.")
             exit 1
-          elsif config_value(:ebs_encrypted) && ! %w{m3.medium m3.large m3.xlarge m3.2xlarge m4.large m4.xlarge
+          elsif config[:ebs_encrypted] && ! %w{m3.medium m3.large m3.xlarge m3.2xlarge m4.large m4.xlarge
```
Comment (tas50) [inline lib/chef/knife/ec2_server_create.rb:661]: wtf is this mess

File: lib/chef/knife/ec2_server_create.rb (hunk 3 above)
Comment (lamont-granquist) [reply lib/chef/knife/ec2_server_create.rb:63]: It has to be this way to be backcompatible with the prior config setting and with the API options, so yes, this code is correct.  The name change here matches the old code in the accumulator.

File: lib/chef/knife/ec2_server_create.rb (hunk 5 above)
Comment (lamont-granquist) [reply lib/chef/knife/ec2_server_create.rb:661]: that's a good question

Comment (lamont-granquist) [reply lib/chef/knife/ec2_server_create.rb:661]: that's just a whitelist of every single AWS instance type that supports "ebs encrypted"

File: lib/chef/knife/ec2_server_create.rb (hunk 4 above)
Comment (lamont-granquist) [reply lib/chef/knife/ec2_server_create.rb:539]: yeah i think it actually still is.

there's still the distinction between `config[:cl_secret]` (set by the command line) and `config[:secret]` (the approved API for the Chef::Config option):

https://github.com/chef/chef/blob/d657a69f13d97f2ae89f39b6632f576b13822c7d/lib/chef/knife/data_bag_secret_options.rb#L65-L75

i went through that a few weeks back and convinced myself i couldn't break that for some reason, but i've blotted it out of my mind by now.

**Pull request merged**